### PR TITLE
[v3] Fix asset auto injection

### PR DESCRIPTION
**Pull request by nuernbergerA**
Fixes [#109 ](https://github.com/livewire/next/issues/109)

### CODE REVIEW

**Review by vmiguellima — APPROVED** [review]
Amazing work!

**Comment by joshhanley** [comment]
@nuernbergerA can you change it to look for the opening `<head>` instead of the closing `</head>`? That way it's most similar to what was happening before, plus then means user defined styles then take preference over Livewire's styles.

File: src/Features/SupportAutoInjectedAssets/UnitTest.php [hunk 1]
```diff
@@ -2,19 +2,127 @@
...
+
+        $this->assertEquals(<<<HTML
+            <html>$livewireStyles
+                <yolo />
+            $livewireScripts</html>
+        HTML, $manipulatedHtml);
+    }
+
+    /** @test */
+    public function it_injects_livewire_assets_wired_formated_html()
```
Comment (zepfietje) [inline src/Features/SupportAutoInjectedAssets/UnitTest.php:61]:
```suggestion
    public function it_injects_livewire_assets_weirdly_formated_html()
```

**Comment by joshhanley** [comment]
Thanks!

**Comment by zepfietje** [comment]
Thanks for handling this, guys! 💜

**Comment by BrandonSurowiec** [comment]
I fooled around with this code to see how it worked, and I'm pretty sure if you have a <header> tag in your template the styles will be injected twice. There will be a need to limit the replacement to one time with the last argument:

```php
->replaceMatches('/(<\s*head[^>]*>)/', '$1'.$livewireStyles, 1)
```

I don't have time to open a PR for it or anything so if you want to look into it, please do.

**Comment by zepfietje** [comment]
Thanks for reporting this, @BrandonSurowiec.
I can confirm this bug occurs indeed.

@nuernbergerA, sorry to mention you, but do you have time to improve this? 🙂

**Comment by nuernbergerA** [comment]
@zepfietje i will open a new PR

**Comment by nuernbergerA** [comment]
Follow up #6020

**Pull request merged**